pkg/rpcbackend/backend.go

```diff
@@ -114,32 +113,41 @@ func (rb *rpcBackend) CallRPC(ctx context.Context, result interface{}, method st
 //
 // In all return paths *including error paths* the RPCResponse is populated
 // so the caller has an RPC structure to send back to the front-end caller.
-func (rb *rpcBackend) SyncRequest(ctx context.Context, rpcReq *RPCRequest) (rpcRes *RPCResponse, err error) {
+func (rc *RPCClient) SyncRequest(ctx context.Context, rpcReq *RPCRequest) (rpcRes *RPCResponse, err error) {
 
 	// We always set the back-end request ID - as we need to support requests coming in from
 	// multiple concurrent clients on our front-end that might use clashing IDs.
 	var beReq = *rpcReq
 	beReq.JSONRpc = "2.0"
-	rb.allocateRequestID(&beReq)
+	rc.allocateRequestID(&beReq)
 	rpcRes = new(RPCResponse)
 
-	log.L(ctx).Infof("RPC:%s:%s --> %s", beReq.ID, rpcReq.ID, rpcReq.Method)
-	res, err := rb.client.R().
+	log.L(ctx).Debugf("RPC:%s:%s --> %s", rpcReq.ID, rpcReq.ID, rpcReq.Method)
+	if logrus.IsLevelEnabled(logrus.TraceLevel) {
+		jsonInput, _ := json.Marshal(rpcReq)
+		log.L(ctx).Tracef("RPC:%s:%s INPUT: %s", rpcReq.ID, rpcReq.ID, jsonInput)
+	}
+	res, err := rc.client.R().
 		SetContext(ctx).
-		SetBody(&beReq).
-		SetResult(rpcRes).
+		SetBody(beReq).
+		SetResult(&rpcRes).
 		SetError(rpcRes).
 		Post("")
+
 	// Restore the original ID
 	rpcRes.ID = rpcReq.ID
 	if err != nil {
 		err := i18n.NewError(ctx, signermsgs.MsgRPCRequestFailed)
-		log.L(ctx).Errorf("RPC:%s:%s <-- ERROR: %s", beReq.ID, rpcReq.ID, err)
-		rpcRes = RPCErrorResponse(err, rpcReq.ID, RPCCodeInternalError)
+		log.L(ctx).Errorf("RPC[%d] <-- ERROR: %s", rpcReq.ID, err)
 		return rpcRes, err
 	}
-	if res.IsError() {
-		log.L(ctx).Errorf("RPC:%s:%s <-- [%d]: %s", beReq.ID, rpcReq.ID, res.StatusCode(), rpcRes.Message())
+	if logrus.IsLevelEnabled(logrus.TraceLevel) {
+		jsonOutput, _ := json.Marshal(rpcRes)
+		log.L(ctx).Tracef("RPC:%s:%s OUTPUT: %s", rpcReq.ID, rpcReq.ID, jsonOutput)
+	}
+	// JSON/RPC allows errors to be returned with a 200 status code, as well as other status codes
+	if res.IsError() || rpcRes.Error != nil && rpcRes.Error.Message != "" {
```

Comment:
The check here needs to be `rpcRes.Error.Code != 0` - that was a recent discovery/change in EVMConnect from testing with various JSON/RPC endpoints. We found one that returned an error, with an empty message.

Reply:
I realized my local `evmconnect` I branched off of was out of date. 🤦‍♂️  Fixed this now.